src/remote-config/remote-config.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseServiceInterface, FirebaseServiceInternalsInterface } from '../firebase-service';
+import { FirebaseApp } from '../firebase-app';
+import {
+  RemoteConfigCondition, RemoteConfigParameter
+} from './remote-config-types';
+import {
+  RemoteConfigApiClient
+} from './remote-config-api-client';
+
+/**
+ * Internals of an RemoteConfig service instance.
+ */
+class RemoteConfigInternals implements FirebaseServiceInternalsInterface {
+  /**
+   * Deletes the service and its associated resources.
+   *
+   * @return {Promise<()>} An empty Promise that will be fulfilled when the service is deleted.
+   */
+  public delete(): Promise<void> {
+    // There are no resources to clean up
+    return Promise.resolve(undefined);
+  }
+}
+
+/**
+ * Remote Config service bound to the provided app.
+ */
+export class RemoteConfig implements FirebaseServiceInterface {
+  public readonly INTERNAL: RemoteConfigInternals = new RemoteConfigInternals();
+
+  private readonly requestHandler: RemoteConfigApiClient;
+
+  /**
+   * @param {FirebaseApp} app The app for this RemoteConfig service.
+   * @constructor
+   */
+  constructor(readonly app: FirebaseApp) {
+    this.requestHandler = new RemoteConfigApiClient(app);
+  }
+
+  /**
```

Comment:
I'd recommend dropping all these placeholder methods from this PR. Makes it easier to review the future PRs as you add these functions.

Reply:
I added the placeholder to get rid of the build error for `requestHandler` as it was unused in the code. :)
Will remove this from the PR and introduce in the next one where we will actually use it to make the requests.

Reply:
Oh you mean remove all the placeholder methods! Make sense. Will update the PR.